libs/agno/agno/tools/whatsapp.py

```diff
@@ -0,0 +1,212 @@
+
+    base_url = "https://graph.facebook.com"
+
+    def __init__(
+        self,
+        access_token: Optional[str] = None,
+        phone_number_id: Optional[str] = None,
+        version: str = "v22.0",
+        recipient_waid: Optional[str] = None,
+    ):
+        """Initialize WhatsApp toolkit.
+
+        Args:
+            access_token: WhatsApp Business API access token
+            phone_number_id: WhatsApp Business Account phone number ID
+            version: API version to use
+            recipient_waid: Default recipient WhatsApp ID (optional)
+        """
+        super().__init__(name="whatsapp")
+
+        # Core credentials
+        self.access_token = access_token or os.getenv("WHATSAPP_ACCESS_TOKEN") or os.getenv("ACCESS_TOKEN")
+        if not self.access_token:
+            logger.error("WHATSAPP_ACCESS_TOKEN not set. Please set the WHATSAPP_ACCESS_TOKEN environment variable.")
+
+        self.phone_number_id = phone_number_id or os.getenv("WHATSAPP_PHONE_NUMBER_ID") or os.getenv("PHONE_NUMBER_ID")
+        if not self.phone_number_id:
+            logger.error(
+                "WHATSAPP_PHONE_NUMBER_ID not set. Please set the WHATSAPP_PHONE_NUMBER_ID environment variable."
+            )
+
+        # Optional default recipient
+        self.default_recipient = recipient_waid or os.getenv("WHATSAPP_RECIPIENT_WAID") or os.getenv("RECIPIENT_WAID")
+
+        # API version
+        self.version = version or os.getenv("WHATSAPP_VERSION") or os.getenv("VERSION", "v22.0")
+
+        # Register methods that can be used by the agent
+        self.register(self.send_text_message_sync)
+        self.register(self.send_template_message_sync)
+
+        # Log configuration status
+        self._log_config_status()
+
+    def _log_config_status(self):
+        """Log the configuration status of the WhatsApp toolkit."""
+        config_status = {
+            "Core credentials": {
+                "access_token": bool(self.access_token),
+                "phone_number_id": bool(self.phone_number_id),
+            },
+            "Optional settings": {"default_recipient": bool(self.default_recipient), "api_version": self.version},
+        }
+        logger.debug(f"WhatsApp toolkit configuration status: {json.dumps(config_status, indent=2)}")
+
+    def _get_headers(self) -> Dict[str, str]:
+        """Get headers for API requests."""
+        return {"Authorization": f"Bearer {self.access_token}", "Content-Type": "application/json"}
+
+    def _get_messages_url(self) -> str:
+        """Get the messages endpoint URL."""
+        return f"{self.base_url}/{self.version}/{self.phone_number_id}/messages"
+
+    async def _send_message_async(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """Send a message asynchronously using the WhatsApp API.
+
+        Args:
+            data: Message data to send
+
+        Returns:
+            API response as dictionary
+        """
+        async with httpx.AsyncClient() as client:
```

Comment:
Should this not have the same logs as sync?

Reply:
added